packages/plugins/other/visitor-plugin-common/src/selection-set-to-object.ts

```diff
@@ -470,28 +525,30 @@ export class SelectionSetToObject<Config extends ParsedDocumentsConfig = ParsedD
       return this.getUnknownType();
     }
 
-    return Object.keys(grouped)
-      .map(typeName => {
-        const relevant = grouped[typeName].filter(Boolean);
+    return (
+      Object.keys(grouped)
+        .map(typeName => {
+          const relevant = grouped[typeName].filter(Boolean);
 
-        if (relevant.length === 0) {
-          return null;
-        } else if (relevant.length === 1) {
-          return relevant[0];
-        } else {
-          return `( ${relevant.join(' & ')} )`;
-        }
-      })
-      .filter(Boolean)
-      .join(' | ');
+          if (relevant.length === 0) {
+            return null;
+          } else if (relevant.length === 1) {
+            return relevant[0];
+          } else {
+            return `( ${relevant.join(' & ')} )`;
+          }
+        })
+        .filter(Boolean)
+        .join(' | ') + this.getEmptyObjectTypeString(mustAddEmptyObject)
```

Comment:
Need to add a test to check whether this is correctly omitted if __typename is selected.